src/server/graphql/models/Team/archiveTeam/archiveTeam.js

```diff
@@ -3,91 +3,64 @@ import {
     const r = getRethink();
     const now = new Date();
 
     // AUTH
-    requireSUOrLead(authToken, updatedTeam.id);
+    requireSUOrLead(authToken, teamId);
     requireWebsocket(socket);
 
-    // VALIDATION
-    const {errors, data: {id, name, isArchived}} = archiveTeamValidation()(updatedTeam);
-    handleSchemaErrors(errors);
-
     // RESOLUTION
-    const notificationTemplate = {
-      id: shortid.generate(),
-      startAt: now,
-      varList: [name]
-    };
     await r.table('Team')
-      .get(id)
-      .pluck('orgId')
-      .do((doc) => ({
-        orgId: doc('orgId'),
-        projectCount: r.table('Project').getAll(id, {index: 'teamId'}).count(),
-        userIds: r.table('TeamMember').getAll(id, {index: 'teamId'})
-        .pluck('userId')
-        .coerceTo('array')
-        .map((newDoc) => newDoc('userId'))
-      }))
-      .do((doc) => ({
-        opType: r.branch(
-          r.and(doc('projectCount').eq(0), doc('userIds').count().eq(1)),
-          TEAM_DELETED,
-          TEAM_ARCHIVED
-        ),
-        orgId: doc('orgId'),
-        userIds: doc('userIds')
+      .get(teamId)
+      .pluck('name', 'orgId')
+      .do((team) => ({
+        projectCount: r.table('Project').getAll(teamId, {index: 'teamId'}).count(),
+        team,
+        userIds: r.table('TeamMember').getAll(teamId, {index: 'teamId'})('userId').coerceTo('array')
       }))
-      .do((doc) => ({
-        opType: doc('opType'),
-        notificationResult: r.table('Notification').insert({
-          ...notificationTemplate,
-          type: doc('opType'),
-          orgId: doc('orgId'),
-          userIds: doc('userIds'),
-        }),
-        teamResult: r.branch(
-          doc('opType').eq(TEAM_ARCHIVED),
-          r.table('Team').get(id).update({isArchived}),
-          r.table('Team').get(id).delete()
-        ),
-        teamMemberResult: r.branch(
-          doc('opType').eq(TEAM_ARCHIVED),
-          null,
-          r.table('TeamMember').getAll(id, {index: 'teamId'}).delete()
-        ),
-        userResult: r.branch(
-          doc('opType').eq(TEAM_ARCHIVED),
-          null,
-          r.table('User').getAll(r.args(doc('userIds'))).forEach((user) =>
-            r.table('User').get(user('id')).update({
+      .do((doc) => r.branch(
+        r.and(doc('projectCount').eq(0), doc('userIds').count().eq(1)),
+        {
+          // Team has no projects nor addn'l TeamMembers, hard delete it:
+          teamResult: r.table('Team').get(teamId).delete(),
+          teamMemberResult: r.table('TeamMember').getAll(teamId, {index: 'teamId'}).delete(),
+          userResult: r.table('User').get(doc('userIds').nth(0))
+            .do((user) =>
               // remove team from user tms, N.B. we don't bother issuing a new token
-              tms: user('tms').difference([id])
-            })
-          )
-        )
-      }));
+              r.table('User').get(user('id')).update({tms: user('tms').difference([teamId])})
```

Comment:
`user` is already a document, no need to look it up again.
I'm not sure why you're only grabbing the first, but assuming it's correct, you can do
`userResult: r.table('User').get(doc('userIds').nth(0)).update((user) => ({tms: user('tms').without(teamId))`